tensorflow_lite_support/python/test/task/audio/audio_classifier_test.py

```diff
@@ -0,0 +1,414 @@
+    model_path = test_util.get_test_data_path(model_name)
+
+    # Creates classifier.
+    if model_file_type is ModelFileType.FILE_NAME:
+      model_file = _ExternalFile(file_name=model_path)
+    elif model_file_type is ModelFileType.FILE_CONTENT:
+      with open(model_path, "rb") as f:
+        model_content = f.read()
+      model_file = _ExternalFile(file_content=model_content)
+    else:
+      # Should never happen
+      raise ValueError("model_file_type is invalid.")
+
+    classifier = self.create_classifier_from_options(
+        model_file, max_results=max_results)
+
+    # Load the input audio file.
+    test_audio_path = test_util.get_test_data_path(audio_file_name)
+    tensor = tensor_audio.TensorAudio.from_wav_file(
+      test_audio_path, classifier.required_input_buffer_size)
+
+    # Classifies the input.
+    audio_result = classifier.classify(tensor)
+    audio_result_dict = json.loads(json_format.MessageToJson(audio_result))
+
+    # Builds test data.
+    expected_result_dict = self.build_test_data(expected_classifications)
+
+    # Comparing results.
+    self.assertDeepAlmostEqual(
+      audio_result_dict, expected_result_dict, delta=_ACCEPTABLE_ERROR_RANGE)
+
+  @parameterized.parameters(
+    (_MULTIHEAD_MODEL_FILE, ModelFileType.FILE_NAME, _TWO_HEADS_AUDIO_FILE,
+     3, _MULTIHEAD_MODEL_CLASSIFICATIONS),
+    (_MULTIHEAD_MODEL_FILE, ModelFileType.FILE_CONTENT, _TWO_HEADS_AUDIO_FILE,
+     3, _MULTIHEAD_MODEL_CLASSIFICATIONS)
+  )
+  def test_classify_multi_head_model(
+      self, model_name, model_file_type, audio_file_name, max_results,
+      expected_classifications):
+    # Get model path.
+    model_path = test_util.get_test_data_path(model_name)
+
+    # Creates classifier.
+    if model_file_type is ModelFileType.FILE_NAME:
+      model_file = _ExternalFile(file_name=model_path)
+    elif model_file_type is ModelFileType.FILE_CONTENT:
+      with open(model_path, "rb") as f:
+        model_content = f.read()
+      model_file = _ExternalFile(file_content=model_content)
+    else:
+      # Should never happen
+      raise ValueError("model_file_type is invalid.")
+
+    classifier = self.create_classifier_from_options(
+      model_file, max_results=max_results)
+    tensor = classifier.create_input_tensor_audio()
+
+    # Load the input audio file.
+    test_audio_path = test_util.get_test_data_path(audio_file_name)
+    tensor = tensor_audio.TensorAudio.from_wav_file(
+      test_audio_path, classifier.required_input_buffer_size)
+
+    # Classifies the input.
+    audio_result = classifier.classify(tensor)
+    audio_result_dict = json.loads(json_format.MessageToJson(audio_result))
+
+    # Builds test data.
+    expected_result_dict = self.build_test_data(expected_classifications)
+
+    # Comparing results.
+    self.assertDeepAlmostEqual(
+      audio_result_dict, expected_result_dict,
+      delta=_ACCEPTABLE_ERROR_RANGE)
+
+  def test_classify_model_on_audio_record_input(self):
```

Comment:
@khanhlvg I've added a test to classify input from an audio recording. The test saves the recording as a file in the home directory in case you want to preview what it sounds like.

Reply:
@kinaryml Thanks! The record.wav sounds correctly so I think the TensorAudio is working correctly with sounddevice. However, the classification results seems incorrect. Could you investigate and fix it?

Reply:
@khanhlvg It seems fine on my end though. I tried testing it with music playing in the background and it classified it as 'Music'.

Reply:
Interesting. On my side, I updated the code to run classification every 1s for a total of 5s, and the output is all the same. But if it works on your side, I'll wait until you've finished implementing the TensorAudio tests and will try again.

```
{'classifications': [{'classes': [{'index': 133, 'score': 0.08203125, 'className': 'Musical instrument'}, {'index': 132, 'score': 0.05859375, 'className': 'Music'}, {'index': 147, 'score': 0.04296875, 'className': 'Keyboard (musical)'}], 'headIndex': 0, 'headName': 'scores'}]}
{'classifications': [{'classes': [{'index': 133, 'score': 0.08203125, 'className': 'Musical instrument'}, {'index': 132, 'score': 0.05859375, 'className': 'Music'}, {'index': 147, 'score': 0.04296875, 'className': 'Keyboard (musical)'}], 'headIndex': 0, 'headName': 'scores'}]}
{'classifications': [{'classes': [{'index': 133, 'score': 0.08203125, 'className': 'Musical instrument'}, {'index': 132, 'score': 0.05859375, 'className': 'Music'}, {'index': 147, 'score': 0.04296875, 'className': 'Keyboard (musical)'}], 'headIndex': 0, 'headName': 'scores'}]}
{'classifications': [{'classes': [{'index': 133, 'score': 0.08203125, 'className': 'Musical instrument'}, {'index': 132, 'score': 0.05859375, 'className': 'Music'}, {'index': 147, 'score': 0.04296875, 'className': 'Keyboard (musical)'}], 'headIndex': 0, 'headName': 'scores'}]}
{'classifications': [{'classes': [{'index': 133, 'score': 0.08203125, 'className': 'Musical instrument'}, {'index': 132, 'score': 0.05859375, 'className': 'Music'}, {'index': 147, 'score': 0.04296875, 'className': 'Keyboard (musical)'}], 'headIndex': 0, 'headName': 'scores'}]}
```

Reply:
@khanhlvg I'm getting the following results if I test using my microphone while running this [track](https://open.spotify.com/track/2loA8ulmO8Ec34aaSqzimz?si=bf7d9bd4c2654e0c) in the background.

```
{'classifications': [{'classes': [{'index': 132, 'score': 0.66796875, 'className': 'Music'}, {'index': 498, 'score': 0.03125, 'className': 'Sound effect'}, {'index': 234, 'score': 0.03125, 'className': 'Electronic music'}], 'headIndex': 0, 'headName': 'scores'}]}
```

Reply:
The [YAMNet](https://github.com/khanhlvg/tflite-support/blob/audio-classifier-python/tensorflow_lite_support/cc/test/testdata/task/audio/yamnet_audio_classifier_with_metadata.tflite) model also produces the same results as [this](https://github.com/khanhlvg/tflite_raspberry_pi/blob/main/audio_classification/test_data/ground_truth.csv)) for the WAV file [meow_16k.wav](https://github.com/khanhlvg/tflite_raspberry_pi/blob/main/audio_classification/test_data/meow_16k.wav).